internal/helm/release/manager.go

```diff
@@ -352,17 +352,10 @@ func createJSONMergePatch(existingJSON, expectedJSON []byte) ([]byte, error) {
 // UninstallRelease performs a Helm release uninstall.
 func (m manager) UninstallRelease(ctx context.Context, opts ...UninstallOption) (*rpb.Release, error) {
 	// Get history of this release
-	h, err := m.storageBackend.History(m.releaseName)
-	if err != nil {
+	if _, err := m.storageBackend.History(m.releaseName); err != nil {
```

Comment:
I think we can remove this check as well. When we call `uninstall.Run()`, Helm will [check history](https://github.com/helm/helm/blob/ab12382ddcd3a27a974b45ed96e47697de221244/pkg/action/uninstall.go#L70). If no release exists, it will return a [wrapped](https://github.com/helm/helm/blob/ab12382ddcd3a27a974b45ed96e47697de221244/pkg/action/uninstall.go#L72) [`driver.ErrReleaseNotFound`](https://github.com/helm/helm/blob/ab12382ddcd3a27a974b45ed96e47697de221244/pkg/storage/driver/secrets.go#L129), which the callee will be able to detect with `errors.Is()`.

TL;DR: This check is duplicative since `uninstall.Run()` calls this already.

Reply:
Rebased. Done as suggested. Tested locally with a normal delete and it seems fine. Tested with a delete with a 0 release and it seems fine as well: 
```
{"level":"info","ts":1611587596.7093048,"logger":"helm.controller","msg":"Release not found, removing finalizer","namespace":"default","name":"nginx-sample","apiVersion":"example.com/v1alpha1","kind":"Nginx","release":"nginx-sample"}
```